sdk/objc/components/capturer/RTCCameraVideoCapturer.m

```diff
@@ -160,17 +173,6 @@ - (void)startCaptureWithDevice:(AVCaptureDevice *)device
                     block:^{
                       RTCLogInfo("startCaptureWithDevice %@ @ %ld fps", format, (long)fps);
 
-#if TARGET_OS_IPHONE
-                      dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Would it be enough to just change these to `dispatch_sync` to ensure the `updateOrientation` gets called? Seems still possible for the threading to get ordered such that the `updateOrientation` call comes in late, even if unlikely.

Also would mean that the capturer doesn't have to turn on the accelerometer all the time.

Reply:
Yes calling beginGeneratingDeviceOrientationNotifications() when not necessary isn't ideal, will make changes.